MiscVBAFunctions/Modules/MiscTables.bas

```diff
@@ -191,3 +191,142 @@ Function TableColumnToArray(TableDicts As Collection, ColumnName As String) As V
     TableColumnToArray = arr
 End Function
 
+
+Function TableColumnToCollection(TableDicts As Collection, ColumnName As String) As Collection
+    ' Append the selected key's value from each Dict in the input Collection to a Collection
+    '
+    ' Args:
+    '   TableDicts: A collection of Dicts.
+    '   ColumnName: Name of the column that will be returned as a Collection.
+    '
+    ' Returns:
+    '   Collection of the selected column.
+    
+    Dim col1 As Collection
+    Dim dict As Dictionary
+    
+    Set col1 = New Collection
+    For Each dict In TableDicts
+        col1.Add dictget(dict, ColumnName)
+    Next dict
+    
+    Set TableColumnToCollection = col1
+End Function
+
+
+Public Sub ResizeLO(SelectedLO As ListObject, NumRows As Long)
+    ' Resize a table to the desired number of data rows. The columns remain unchanged.
+    ' If NumRows is set to "0", the table will instead be resized to "1" and the
+    ' content in that 1st row will be cleared.
+    '
+    ' Args:
+    '   SelectedLO: Selected List Object.
+    '   NumRows: Number of desired rows.
+    
+    Dim oldNumRows As Long
+    oldNumRows = SelectedLO.ListRows.Count
+    
+    ' Don't allow tables to be resized to zero rows.  Excel won't really do
+    ' this anyway - it's possible to get the DataBodyRange to be empty, in
+    ' which case the single row displayed in the table is the Insert row, but
+    ' this creates more problems than it solves.  Instead, resize the table to
+    ' one row, and set a flag to clear out any remaining data later.
+    Dim clearTable As Boolean
+    clearTable = False
+    If NumRows = 0 Then
+        NumRows = 1
+        clearTable = True
+    End If
+    
+    ' Resize the table (add 1 to the number of rows because mListObject.Range
+    ' includes the header row).
+    SelectedLO.Resize _
+        SelectedLO.Range.Resize( _
+            NumRows + 1, _
+            SelectedLO.ListColumns.Count)
+    
+    ' If the table is resized to have one row, but the row contains no data,
+    ' the row will be treated as the Insert row, and the data row count will
+    ' remain zero.  This will cause problems since the table doesn't actually
```

Comment:
This is what we want, if a table doesn't have any data it should not have a DataBodyRange